vars/build.groovy

```diff
@@ -49,21 +52,27 @@ def call(Map args) {
   }
 }
 
-def createImageStream(imageStreams, namespace) {
-    imageStreams.each { imageStream ->
-        def isName = imageStream.metadata.name
-        def isFound = Utils.shWithOutput(this, "oc get is/$isName -n $namespace --ignore-not-found")
-        if (!isFound) {
-          Utils.ocApply(this, imageStream, namespace)
-        } else {
-          echo "image stream exist ${isName}"
-        }
+def createImageStream(imageStreams, ns) {
+    imageStreams.each { is ->
+      ocApply(this, is, ns)
     }
 }
 
-def buildProject(buildConfigs, namespace) {
+def buildProject(buildConfigs, ns) {
     buildConfigs.each { buildConfig ->
-        Utils.ocApply(this, buildConfig, namespace)
-        openshiftBuild(buildConfig: "${buildConfig.metadata.name}", showBuildLogs: 'true')
+      startBuild(buildConfig, ns)
+      watchBuildLogs(buildConfig, ns)
     }
 }
+
+def startBuild(buildConfig, ns) {
+  ocApply(this, buildConfig, ns)
+  sh "oc start-build ${buildConfig.metadata.name} -n $ns"
+}
+
+def watchBuildLogs(buildConfig, ns) {
+  retry(3) {
+    sleep 3
```

Comment:
why the sleep ? oc should wait until its coming isnt it ?

Reply:
It may not be guaranteed to build pod get to start it could fail oc logs bc command 
Hence adding some pause. It's not a nice way though but could work for the time being.

Reply:
@sthaha just tagging sunil so he can be aware it,

Reply:
:+1:

Reply:
@hrishin magic `sleep 3` worries me as well. what is the intention here?

Reply:
should we rather accept a timeout from user (defaults to 3 or whatever) which is the max timeout for builds to happen or fail the build/pipeline?

Reply:
Yes, @sthaha for both point. 
Lets avoid the problem. Adding `-F` in  `oc start-build` command.

` -F, --follow=false: Start a build and watch its logs until it completes or fails`